### PR TITLE
mailbox documentation update

### PR DESCRIPTION
**Pull request by xuhz**
Sonal, the mailbox documentation is updated as you requested, pls review

### CODE REVIEW

File: src/runtime_src/doc/toc/cloud_vendor_support.rst [hunk 1]
```diff
@@ -191,3 +191,14 @@ The following picture illustrates how XRT is being deployed in different scenari
 
 .. image:: xrt-deployment-cloud.svg
    :align: center
+
+Special Case
+============
+
+There is special case where download xclbin from within user VM is not required.
+
+In this special case, neither MSD/MPD nor plugins are required since there is no real xclbin download happening. xclbins can be preloaded
```
Comment (sonals) [inline src/runtime_src/doc/toc/cloud_vendor_support.rst:200]: there is no real xclbin download happening --> no xclbin download is allowed from guest.

File: src/runtime_src/doc/toc/cloud_vendor_support.rst [hunk 2]
```diff
@@ -191,3 +191,14 @@ The following picture illustrates how XRT is being deployed in different scenari
...
+either by hypervisor or by a service domain (VM) where the mgmt PF is assigned. The apps in user VM are running without any change, ie
```
Comment (sonals) [inline src/runtime_src/doc/toc/cloud_vendor_support.rst:201]: service domain (VM) --> or a dom0 type VM

 are running without any change --> run without any change

ie -> i.e.

File: src/runtime_src/doc/toc/cloud_vendor_support.rst [hunk 3]
```diff
@@ -191,3 +191,14 @@ The following picture illustrates how XRT is being deployed in different scenari
...
+xclbin download ioctl is still issued to xocl driver, xocl driver gets the uuid of the running xclbin with a XCL_MAILBOX_REQ_PEER_DATA
```
Comment (sonals) [inline src/runtime_src/doc/toc/cloud_vendor_support.rst:202]: running xclbin --> preloaded xclbin

File: src/runtime_src/doc/toc/cloud_vendor_support.rst [hunk 4]
```diff
@@ -191,3 +191,14 @@ The following picture illustrates how XRT is being deployed in different scenari
...
+mailbox opcode to xclmgmt, and if the uuid matches the one of the xclbin being downloaded, the ioctl returns success without real xclbin
```
Comment (sonals) [inline src/runtime_src/doc/toc/cloud_vendor_support.rst:203]: the one of the xclbin being downloaded, the ioctl returns success  --> with that of the xclbin requested for download, the ioctl returns immediately with success. If the uuids do not match download request in the guest fails.

File: src/runtime_src/doc/toc/fpga_device_ready.rst [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+FPGA device readiness within user VM
+************************************
+
+To make the FPGA in user VM ready to use, the mgmt side has to be ready first. Beside that, depending on how
+the FPGA is deployed, there are still other factors to be considered.
+
+With MSD/MPD or Plugins used
+============================
+
+In this case, :ref:`mailbox.main.rst` has to be opened from user, and  MPD service must be active.
```
Comment (sonals) [inline src/runtime_src/doc/toc/fpga_device_ready.rst:13]: user --> user VM

MPD service must be active --> MPD service must be active in the guest VM.

File: src/runtime_src/doc/toc/fpga_device_ready.rst [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        # where 0x100 is 1 << XCL_MAILBOX_REQ_LOAD_XCLBIN, as defined as below
+        # XCL_MAILBOX_REQ_UNKNOWN =		0,
+        # XCL_MAILBOX_REQ_TEST_READY =		1,
+        # XCL_MAILBOX_REQ_TEST_READ =		2,
+        # XCL_MAILBOX_REQ_LOCK_BITSTREAM =	3,
+        # XCL_MAILBOX_REQ_UNLOCK_BITSTREAM =	4,
+        # XCL_MAILBOX_REQ_HOT_RESET =		5,
+        # XCL_MAILBOX_REQ_FIREWALL =		6,
+        # XCL_MAILBOX_REQ_LOAD_XCLBIN_KADDR =	7,
+        # XCL_MAILBOX_REQ_LOAD_XCLBIN =		8,
```
Comment (sonals) [inline src/runtime_src/doc/toc/fpga_device_ready.rst:42]: Align 8 with the rest of the entries in the column

File: src/runtime_src/doc/toc/fpga_device_ready.rst [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+For MSD/MPD, the MSD service must be active also either from hypervisor or mgmt service domain (VM)
+
+For Plugins, the vendor specific plugin package has to be installed
+
+Without MSD/MPD or Plugins used
+===============================
+  
+In this case, :ref:`mailbox.main.rst` is still being used, but there is no open to the mailbox subdevice from user,
+and MPD service is inactive.
+To make the FPGA device ready, the XCL_MAILBOX_REQ_LOAD_XCLBIN opcode has to be disabled from xclmgmt like following,
```
Comment (sonals) [inline src/runtime_src/doc/toc/fpga_device_ready.rst:24]: disabled from xclmgmt --> disabled in xclmgm

File: src/runtime_src/doc/toc/security.rst [hunk 8]
```diff
@@ -187,10 +187,13 @@ partition is not fully AXI-compliant or deadlocks/stalls/hangs during operation.
 partition fails, AXI Firewall *trips* -- it starts completing AXI transactions on behalf of the slave so the
 master and the specific AXI bus is not impacted -- to protect the Shell. The AXI Firewall starts completing
 all transactions on behalf of misbehaving slave while also notifying the mgmt driver about the trip. The
-xclmgmt driver then starts taking recovery action. xclmgmt posts a message to xocl using MailBox to inform
-the peer about FireWall trip. xocl can suggest a reset by sending a reset command to xclmgmt on MailBox. Note
-that even if no reset is performed the AXI Protocol Firewall will continue to protect the host PCIe bus. DFX
-partition will be unavailable till device is reset. **A reboot of host is not required to reset the device.**
+xclmgmt driver then starts taking recovery action. xclmgmt posts a XCL_MAILBOX_REQ_FIREWALL message to xocl using
+MailBox to inform the peer about FireWall trip. xocl can suggest a reset by sending a XCL_MAILBOX_REQ_HOT_RESET message
+to xclmgmt on MailBox. Note that even if no reset is performed the AXI Protocol Firewall will continue to protect the host PCIe bus.
```
Comment (sonals) [inline src/runtime_src/doc/toc/security.rst:192]: to xclmgmt on MailBox --> o xclmgmt via MailBox

File: src/runtime_src/doc/toc/security.rst [hunk 9]
```diff
@@ -187,10 +187,13 @@ partition is not fully AXI-compliant or deadlocks/stalls/hangs during operation.
...
+DFX partition will be unavailable till device is reset. **A reboot of host is not required to reset the device.**
+
+On board that has multiple FPGA devices on one board and supports card level reset like u30, message exchanges on mailbox are exactly
```
Comment (sonals) [inline src/runtime_src/doc/toc/security.rst:195]: On board that has multiple FPGA devices on one board and supports card level reset like u30 --> Alveo boards with multiple FPGA devices on the same board like U30 support card level. 

message exchanges on mailbox are exactly the same to single FPGA board --> Mailbox usage by each device on the card is similar to that of single device cards, however firewall trip in device will trigger reset all devices on the card.

File: src/runtime_src/doc/toc/security.rst [hunk 10]
```diff
@@ -279,8 +282,51 @@ Mailbox
 
 Mailbox is used for communication between user physical function driver, xocl and management physical
 function driver, xclmgmt. The Mailbox hardware design and xclmgmt driver mailbox handling implementation
-has the ability to throttle requests coming from xocl driver which protects it from a potential DoS
-initiated by a malicious xocl driver. :ref:`mailbox.main.rst` has details on mailbox usage.
+has the ability to throttle requests coming from xocl driver. 
+
+xclmgmt driver has twofold security protections on the h/w mailbox. From packet layer, xclmgmt monitors 
+the receiving packet rates and set a threshold. If the receiving packet rates exceeds the threshold, a
```
Comment (sonals) [inline src/runtime_src/doc/toc/security.rst:288]: and set a threshold. --> and can enforce a threshold.

File: src/runtime_src/doc/toc/security.rst [hunk 11]
```diff
@@ -279,8 +282,51 @@ Mailbox
...
+malicious behavior, ie, DoS is considered occurring from xocl, and the whole mailbox will be disabled. Only
```
Comment (sonals) [inline src/runtime_src/doc/toc/security.rst:289]: exceeds the threshold, a malicious behavior, ie, DoS is considered occurring from xocl, and the whole mailbox will be disabled --> exceeds the threshold, the mailbox is disabled which prevents the guest from sending any more commands over mailbox.

File: src/runtime_src/doc/toc/security.rst [hunk 12]
```diff
@@ -279,8 +282,51 @@ Mailbox
...
+a hot reset on the FPGA device from xclmgmt can recover it. Frow message layer, xclmgmt can sets all the
+mailbox opcodes it doesn't expect to see. If seen, they will be discarded.
+
+Admin can set the disabled mailbox opcodes like following,
```
Comment (sonals) [inline src/runtime_src/doc/toc/security.rst:293]: Admin can set the disabled mailbox opcodes like following, --> System administrator can configure the mailbox to ignore specific opcodes using xbmgmt. Here is an example.

File: src/runtime_src/doc/toc/security.rst [hunk 13]
```diff
@@ -279,8 +282,51 @@ Mailbox
...
 Mailbox is used for communication between user physical function driver, xocl and management physical
 function driver, xclmgmt. The Mailbox hardware design and xclmgmt driver mailbox handling implementation
-has the ability to throttle requests coming from xocl driver which protects it from a potential DoS
-initiated by a malicious xocl driver. :ref:`mailbox.main.rst` has details on mailbox usage.
+has the ability to throttle requests coming from xocl driver. 
+
+xclmgmt driver has twofold security protections on the h/w mailbox. From packet layer, xclmgmt monitors 
+the receiving packet rates and set a threshold. If the receiving packet rates exceeds the threshold, a
+malicious behavior, ie, DoS is considered occurring from xocl, and the whole mailbox will be disabled. Only
+a hot reset on the FPGA device from xclmgmt can recover it. Frow message layer, xclmgmt can sets all the
```
Comment (sonals) [inline src/runtime_src/doc/toc/security.rst:290]: Frow message layer, xclmgmt can sets all the mailbox opcodes it doesn't expect to see. If seen, they will be discarded. --> System administrator can configure the xclmgmt driver to ignore specific mailbox opcodes. In this case the xclmgmt driver will ignore those opcodes.

**Review by sonals — CHANGES_REQUESTED** [review]
Should we not have only example for xbmgmt to disable opcodes?

Other suggested changes below.

File: src/runtime_src/doc/toc/cloud_vendor_support.rst (hunk 1 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/cloud_vendor_support.rst]: changed

File: src/runtime_src/doc/toc/cloud_vendor_support.rst (hunk 2 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/cloud_vendor_support.rst]: changed

File: src/runtime_src/doc/toc/cloud_vendor_support.rst (hunk 3 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/cloud_vendor_support.rst]: changed

File: src/runtime_src/doc/toc/cloud_vendor_support.rst (hunk 4 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/cloud_vendor_support.rst]: changed

File: src/runtime_src/doc/toc/fpga_device_ready.rst (hunk 5 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/fpga_device_ready.rst]: changed

File: src/runtime_src/doc/toc/fpga_device_ready.rst (hunk 6 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/fpga_device_ready.rst]: changed

File: src/runtime_src/doc/toc/fpga_device_ready.rst (hunk 7 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/fpga_device_ready.rst]: changed

File: src/runtime_src/doc/toc/security.rst (hunk 8 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/security.rst]: changed

File: src/runtime_src/doc/toc/security.rst (hunk 9 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/security.rst]: changed

File: src/runtime_src/doc/toc/security.rst (hunk 10 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/security.rst]: changed

File: src/runtime_src/doc/toc/security.rst (hunk 11 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/security.rst]: changed

File: src/runtime_src/doc/toc/security.rst (hunk 12 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/security.rst]: changed

File: src/runtime_src/doc/toc/security.rst (hunk 13 above)
Comment (xuhz) [reply src/runtime_src/doc/toc/security.rst]: changed

**Pull request merged**